docs/analyse.md

```diff
@@ -40,10 +40,307 @@ watch --interval 30 python -m neps.status ROOT_DIRECTORY
+        lr=neps.FloatParameter(lower=1e-5, upper=1e-1, log=True),
+        batch_size=neps.IntegerParameter(lower=32, upper=256),
+        weight_decay=neps.FloatParameter(lower=1e-4, upper=1e-1, log=True),
+    )
+
+    return search_space
+
+
+# Designing the run_pipeline
+def run_pipeline(**config) -> dict:
+    # Create the model
+    model = MLP()
+
+    # Epochs to train the model, can be parameterized as fidelity
+    max_epochs = 2
+
+    # Load the MNIST dataset for training.
+    train_loader, _ = MNIST(batch_size=config["batch_size"])
+
+    # Define the optimizer, criterion, and a scheduler for the learning rate
+    optimizer = torch.optim.Adam(
+        model.parameters(),
+        lr=config["lr"],
+        weight_decay=config["weight_decay"],
+    )
+    criterion = nn.CrossEntropyLoss()
+    scheduler = lr_scheduler.StepLR(optimizer, step_size=1, gamma=0.75)
+
+    # Create the training loop
+    for i in range(max_epochs):
+        total_loss = 0.0
+        incorrect_images = []
+        for x, y in train_loader:
+            optimizer.zero_grad()
+            output = model(x)
+            loss = criterion(output, y)
+            loss.backward()
+            optimizer.step()
+
+            predicted_labels = torch.argmax(output, dim=1)
+            incorrect_mask = predicted_labels != y
+            incorrect_images.append(x[incorrect_mask])
+
+            total_loss += loss.item()
+
+        # Calculate average loss for the epoch
+        avg_loss = total_loss / len(train_loader)
+
+        if len(incorrect_images) > 0:
+            incorrect_images = torch.cat(incorrect_images, dim=0)
+
+        ################## TensorBoard Logging Start ##################
+        tblogger.log(
+            loss=avg_loss,
+            current_epoch=i,
+            write_summary_incumbent=True,
+            writer_config_scalar=True,
+            writer_config_hparam=True,
+            # Appending extra data
+            extra_data={
+                "lr_decay": tblogger.scalar_logging(
+                    value=scheduler.get_last_lr()[0]
+                ),
+                "miss_img": tblogger.image_logging(
+                    image=incorrect_images, seed=2
+                ),
+            },
+        )
+        ################## TensorBoard Logging End ##################
+        scheduler.step()
+        
+        print(f"  Epoch {i + 1} / {max_epochs} Train Error: {avg_loss} ")
+
+    # Retrieve and return the information from the run pipeline.
+    return {
+        "loss": loss,
+        "info_dict": {
+            "cost": max_epochs,
+        },
+    }
+
+
+# Running the search with random search.
+if __name__ == "__main__":
+    set_seed(112)
+    logging.basicConfig(level=logging.INFO)
+
+    neps.run(
+        run_pipeline=run_pipeline,
+        pipeline_space=pipeline_space(),
+        root_directory="results/mnist_logging_example",
+        searcher="random_search",
+        max_evaluations_total=5,
+    )
+
+```
```

Comment:
This feels way too much content on one page of documentation, especially something that the user HAS to scroll through.
Don't we have an example? Or we can move this as an example and link it here.

Reply:
We do have an example. We could link the example here as the guide and showcase the images only.